src/CMakeLists.txt

```diff
@@ -135,136 +135,151 @@ if(DLAF_WITH_PRECOMPILED_HEADERS)
+  factorization
+  SOURCES factorization/cholesky/mc.cpp $<$<BOOL:${DLAF_WITH_GPU}>:factorization/cholesky/gpu.cpp>
+          factorization/qr/mc.cpp $<$<BOOL:${DLAF_WITH_GPU}>:factorization/qr/gpu.cpp>
+  LIBRARIES dlaf.core
 )
-target_link_libraries(dlaf.factorization PUBLIC dlaf.prop)
-target_add_warnings(dlaf.factorization)
-DLAF_addPrecompiledHeaders(dlaf.factorization)
 
 # Define DLAF's multiplication library
-add_library(dlaf.multiplication OBJECT
-  multiplication/general/mc.cpp
-  $<$<BOOL:${DLAF_WITH_GPU}>:multiplication/general/gpu.cpp>
-  multiplication/triangular/mc.cpp
-  $<$<BOOL:${DLAF_WITH_GPU}>:multiplication/triangular/gpu.cpp>)
-target_link_libraries(dlaf.multiplication PUBLIC dlaf.prop)
-target_add_warnings(dlaf.multiplication)
-DLAF_addPrecompiledHeaders(dlaf.multiplication)
+DLAF_addSublibrary(
+  multiplication
+  SOURCES multiplication/general/mc.cpp $<$<BOOL:${DLAF_WITH_GPU}>:multiplication/general/gpu.cpp>
+          multiplication/triangular/mc.cpp $<$<BOOL:${DLAF_WITH_GPU}>:multiplication/triangular/gpu.cpp>
+  LIBRARIES dlaf.core
+)
 
 # Define DLAF's permutations library
-add_library(
-  dlaf.permutations OBJECT permutations/general/mc.cpp
-                           $<$<BOOL:${DLAF_WITH_GPU}>:permutations/general/gpu.cpp>
-                           $<$<BOOL:${DLAF_WITH_GPU}>:permutations/general/perms.cu>
+DLAF_addSublibrary(
+  permutations
+  SOURCES permutations/general/mc.cpp $<$<BOOL:${DLAF_WITH_GPU}>:permutations/general/gpu.cpp>
+  GPU_SOURCES permutations/general/perms.cu
+  LIBRARIES dlaf.core
 )
-if (DLAF_WITH_HIP)
-    set_source_files_properties(permutations/general/perms.cu PROPERTIES LANGUAGE HIP)
-endif()
-target_link_libraries(dlaf.permutations PUBLIC dlaf.prop)
-target_add_warnings(dlaf.permutations)
-DLAF_addPrecompiledHeaders(dlaf.permutations)
 
 # Define DLAF's solver library
-add_library(dlaf.solver OBJECT
-  solver/triangular/mc.cpp
-  $<$<BOOL:${DLAF_WITH_GPU}>:solver/triangular/gpu.cpp>)
-target_link_libraries(dlaf.solver PUBLIC dlaf.prop)
-target_add_warnings(dlaf.solver)
-DLAF_addPrecompiledHeaders(dlaf.solver)
+DLAF_addSublibrary(
+  solver SOURCES solver/triangular/mc.cpp $<$<BOOL:${DLAF_WITH_GPU}>:solver/triangular/gpu.cpp>
+  LIBRARIES dlaf.core
+)
+
+target_compile_options(
+  dlaf.core_object PRIVATE $<$<CXX_COMPILER_ID:GNU>:-Wno-error=stringop-overflow
+                           -Wno-error=array-bounds>
+)
+target_compile_options(dlaf.eigensolver_object PRIVATE $<$<CXX_COMPILER_ID:GNU>:-Wno-error=array-bounds>)
+target_compile_options(
+  dlaf.tridiagonal_eigensolver_object PRIVATE $<$<CXX_COMPILER_ID:GNU>:-Wno-error=array-bounds>
+)
+target_compile_options(
+  dlaf.factorization_object PRIVATE $<$<CXX_COMPILER_ID:GNU>:-Wno-error=array-bounds>
+)
+target_compile_options(
+  dlaf.multiplication_object PRIVATE $<$<CXX_COMPILER_ID:GNU>:-Wno-error=array-bounds>
+)
+target_compile_options(dlaf.solver_object PRIVATE $<$<CXX_COMPILER_ID:GNU>:-Wno-error=array-bounds>)
```

Comment:
These warnings are disabled due to warnings coming from pika itself. https://github.com/eth-cscs/DLA-Future/issues/688 is a reminder to reenable them after pika 0.10.0, which will ignore the warnings in pika itself (https://github.com/pika-org/pika/pull/489). As far as I can tell these warnings are false positives (explanation in https://github.com/pika-org/pika/pull/489).